_posts/2024-10-15-Announcing-Data-Prepper-2.10.0.md

```diff
@@ -0,0 +1,129 @@
+        index: "${getMetadata(\"opensearch_index\")}"
+        action: "${getMetadata(\"opensearch_action\")}"
+        document_id: "${getMetadata(\"opensearch_id\")}"
+        routing: "${getMetadata(\"opensearch_routing\")}"
+        pipeline: "${getMetadata(\"opensearch_pipeline\")}"
+```
+
+Consider the sample:
+
+```
+POST _bulk
+{ "index": { "_index": "movies", "_id": "tt1979320" } }
+{ "title": "Rush", "year": 2013 }
+```
+
+The above request will be ingested into OpenSearch and a new document will be created under the index movies with a document id `tt1979320` with the document `{ "title": "Rush", "year": 2013 }`.
+
+Over time, the Data Prepper maintainers are interested in expanding this source to support other indexing APIs to allow it to stand-in for an OpenSearch cluster for ingestion workloads.
+To learn more or express interest see [Provide an OpenSearch API source #4180](https://github.com/opensearch-project/data-prepper/issues/4180).
+
+
+## Kinesis source
+
+[Amazon Kinesis Data Streams](https://docs.aws.amazon.com/streams/latest/dev/introduction.html) is a high speed streaming data service. 
+Data Prepper is introducing a new source named `kinesis` which can be used to ingest stream records data from multiple Kinesis data streams into OpenSearch clusters. 
+You can configure it to read stream records from the beginning or from the latest record. 
+Moreover, if you enable end to end acknowledgements, Kinesis data streams will be checkpointed to prevent duplicate processing of records.
+
+Sample pipeline
+
+```
+version: "2"
+kinesis-pipeline:
+  source:
+    kinesis:
+      codec:
+        newline:
+      streams:
+        - stream_name: "MyStream1"
+          initial_position: LATEST
+          checkpoint_interval: "PT5M"
+        - stream_name: "MyStream2"
+          # Enable this if ingestion should start from the start of the stream.
+          initial_position: EARLIEST
+          consumer_strategy: "polling"
+          polling:
+            max_polling_records: 100
+            idle_time_between_reads: "250ms"
+```
+
+
+
+## Other features and improvements
+
+Data Prepper 2.10 has a number of other changes to make it more powerful for the community.
+
+* The kafka source now supports authenticating with an Apache Kafka cluster using SASL/SCRAM in addition to the SASL/PLAIN authentication provided in previous versions.
+* Data Prepper can now parse OpenTelemetry logs from sources such as Amazon S3. The new `otel_logs` codec parses data from OpenTelemetry Protocol (OTLP) JSON formatted files. Now you can write OpenTelemetry logs from [AWS S3 Exporter for OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awss3exporter/README.md) and read these from Data Prepper. 
+* Additionally, the maintainers have worked to improve the performance through the addition of an internal cache for event keys. Data Prepper administrators can configure this cache as necessary. 
+
```

Comment:
Line 115: The name of the service is "Amazon S3" (never "AWS S3").

Reply:
On line 115, I'm using the exact name of the OTel plugin: "AWS S3 Exporter for OpenTelemetry Collector"

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awss3exporter/README.md

Reply:
Ah, I see. In that case, we can leave as is.